core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -153,47 +153,43 @@ public TableScan asOfTime(long timestampMillis) {
 
   @Override
   public TableScan option(String property, String value) {
-    ImmutableMap.Builder<String, String> builder = ImmutableMap.builder();
-    builder.putAll(context.options());
-    builder.put(property, value);
-
     return newRefinedScan(
-        ops, table, schema, context.options(builder.build()));
+        ops, table, schema, context.withOption(property, value));
   }
 
   @Override
   public TableScan project(Schema projectedSchema) {
     return newRefinedScan(
-        ops, table, projectedSchema, context.copy());
+        ops, table, projectedSchema, context);
   }
 
   @Override
   public TableScan caseSensitive(boolean scanCaseSensitive) {
     return newRefinedScan(
-        ops, table, schema, context.caseSensitive(scanCaseSensitive));
+        ops, table, schema, context.isCaseSensitive(scanCaseSensitive));
```

Comment:
Using the prefix `is` normal signals a boolean getter, not a setter. What about `setCaseSensitive` or just `caseSensitive`?